src/Kiota.Builder/PathSegmenters/CSharpPathSegmenter.cs

```diff
@@ -1,11 +1,23 @@
-﻿using Kiota.Builder.CodeDOM;
+﻿using System.IO;
+using Kiota.Builder.CodeDOM;
 using Kiota.Builder.Extensions;
 
 namespace Kiota.Builder.PathSegmenters;
 public class CSharpPathSegmenter : CommonPathSegmenter
 {
     public CSharpPathSegmenter(string rootPath, string clientNamespaceName) : base(rootPath, clientNamespaceName) { }
     public override string FileSuffix => ".cs";
-    public override string NormalizeNamespaceSegment(string segmentName) => segmentName.ToFirstCharacterUpperCase();
+    public override string NormalizeNamespaceSegment(string segmentName) => segmentName.ToFirstCharacterUpperCase().ShortenFileName(64);//don't namespace folders get too long(longer than its hash)
     public override string NormalizeFileName(CodeElement currentElement) => GetLastFileNameSegment(currentElement).ToFirstCharacterUpperCase();
+    public override string NormalizePath(string fullPath)
+    {
+        if (ExceedsMaxPathLength(fullPath) && Path.GetDirectoryName(fullPath) is string directoryName)
+        {
+            var availableLength = MaxFilePathLength - (directoryName.Length + FileSuffix.Length);
+            return Path.Combine(directoryName, Path.GetFileName(fullPath).ShortenFileName(availableLength)) + FileSuffix;
+        }
+        return fullPath;
+    }
+    private const int MaxFilePathLength = 256;//windows max path length
```

Comment:
I believe we're using a longer limit for other languages (Go comes to mind), it maybe worth aligning things across the board if we can.
260 seems short, that's windows without long path, which windows has had enabled by default for a while now AFAIK, maybe we can bump this a little

Reply:
Sorry about this. I read some docs wrong. The `256` limit is per folder/file name but the entire filepath should get up to 32K. 
Updated the relevant pathsegmenters to limit the folder name to be same across and moved up the file normalization to up to the base virtual function so that it can be overridden if need be with a max length of 32K.